### PR TITLE
Check heap pointer in Scatter creation policy

### LINKED ISSUE #115: Scatter policy does not check heap for NULL pointer
As described in #111 and #110 it is possible that a NULL pointer is passed to the scatter policy with a size != 0. That policy would then use the pointer, assuming it to be valid without any further checks eventually leading to a crash.

### PR DESCRIPTION
**Pull request by slizzered**
fix #115

### CODE REVIEW

**Comment by slizzered** [comment]
don't merge yet. there appear to be some warnings

File: src/include/mallocMC/creationPolicies/Scatter_impl.hpp [hunk 1]
```diff
@@ -817,6 +818,13 @@ namespace ScatterKernelDetail{
 
       template < typename T_DeviceAllocator >
       static void* initHeap( T_DeviceAllocator* heap, void* pool, size_t memsize){
+        if( pool == NULL && memsize != 0 )
+        {
+          throw std::invalid_argument(
+            "Scatter policy cannot use NULL for non-empty memory pools.\
```
Comment (ax3l) [inline src/include/mallocMC/creationPolicies/Scatter_impl.hpp]: Should be newlines as
```C++
    "something. "
    "New sentence."

Comment (slizzered) [inline src/include/mallocMC/creationPolicies/Scatter_impl.hpp]: Not sure I understand your point. The constructor of `invalid_argument` only takes a single string literal as input. The sentences were only split into 2 lines to comply with the 80 character limit per line.

I believe the final output should be on a single line to make clear that the sentences together and originate at the same place.

Comment (ax3l) [inline src/include/mallocMC/creationPolicies/Scatter_impl.hpp]: the point is: the syntax I gave you is the correct way to split *a single* string argument.

Comment (slizzered) [inline src/include/mallocMC/creationPolicies/Scatter_impl.hpp]: Wow, I was actually not aware of the correct way to split a string in C++. Thanks for your input! :sparkling_heart:

Comment (ax3l) [inline src/include/mallocMC/creationPolicies/Scatter_impl.hpp]: :sparkles:

**Comment by slizzered** [comment]
ok, can be merged now :)

**Comment by ax3l** [comment]
cool, thanks!

**Pull request merged**